closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -0,0 +1,1044 @@
+         * {Policy,
+         * JavaPolicy, *}, {CertPathValidator, *, *}].
+         */
+        private void initConstraints() {
+
+            for (int pNum = 1; pNum <= providersSN.size(); pNum++) {
+
+                String providerName = providersSN.get(pNum - 1);
+                String providerInfo = securityProps
+                        .getProperty(propsPrefix + ".jce.provider." + pNum);
+
+                if (debug != null) {
+                    debug.println("Loading constraints for security provider: " + providerName);
+                }
+
+                // Check if the provider has constraints
+                if (providerInfo.indexOf('[') < 0 && providerInfo.indexOf(']') < 0) {
+                    if (debug != null) {
+                        debug.println("No constraints for security provider: " + providerName);
+                    }
+                    continue;
+                }
+
+                // Remove the whitespaces in the format separator if there are.
+                providerInfo = providerInfo.trim()
+                        .replaceAll("\\[\\s*\\{", "[{")
+                        .replaceAll("\\}\\s*\\]", "}]")
+                        .replaceAll("\\}\\s*,\\s*\\{", "},{");
+
+                int startIndex = providerInfo.indexOf("[{");
+                int endIndex = providerInfo.indexOf("}]");
+
+                // Provider with constraints.
+                if ((startIndex > 0) && (endIndex > 0)) {
+                    String[] constrArray = providerInfo
+                            .substring(startIndex + 2, endIndex).split("\\},\\{");
+
+                    if (constrArray.length <= 0) {
+                        printStackTraceAndExit("Constraint format is incorrect: " + providerInfo);
+                    }
+
+                    // Constraint object array.
+                    // For each constraint type, algorithm and attributes.
+                    Constraint[] constraints = new Constraint[constrArray.length];
+
+                    int cNum = 0;
+                    for (String constr : constrArray) {
+                        String[] input = constr.split(",");
+
+                        // Each constraint must includes 3 fields(type, algorithm, attributes).
+                        if (input.length != 3) {
+                            printStackTraceAndExit("Constraint format is incorrect: " + providerInfo);
+                        }
+
+                        String inType = input[0].trim();
+                        String inAlgorithm = input[1].trim();
+                        String inAttributes = input[2].trim();
+
+                        // Each attribute must includes 2 fields(key and value) or *.
+                        if (!isAsterisk(inAttributes)) {
+                            String[] attributeArray = inAttributes.split(":");
+                            for (String attribute : attributeArray) {
+                                String[] in = attribute.split("=");
```

Comment:
I suggest this use of `split()` (and the one on line 778) should be `split("=", 2)`, otherwise some of the input could be ignored.